File: server/tools/check_outages.mjs
#!/usr/bin/env node

async function main(){
  try{
    const res = await fetch('http://localhost:7654/api/reports/debug/outages')
    const j = await res.json()
    const outages = j.outages || []
    const total = j.totalOutages ?? j.total ?? outages.length
    const sample = outages.slice(0,5)
    console.log(JSON.stringify({ total, sample }))
  } catch(e){
    console.error(JSON.stringify({ error: String(e) }))
    process.exit(2)
  }
}
main()

File: server/tools/check_outages.py
#!/usr/bin/env python3
import sys, json, urllib.request

url = 'http://localhost:7654/api/reports/debug/outages'
try:
    with urllib.request.urlopen(url, timeout=20) as r:
        data = json.load(r)
except Exception as e:
    print(json.dumps({'error': str(e)}))
    sys.exit(2)

outages = data.get('outages') or []
total = data.get('totalOutages', data.get('total', len(outages)))
sample = outages[:5]
print(json.dumps({'total': total, 'sample': sample}, ensure_ascii=False))
